packages/features/insights/filters/routing/FilterBar.tsx

```diff
@@ -0,0 +1,76 @@
+import type { RoutingFormTableType } from "@calcom/features/insights/components/RoutingFormResponsesTable";
+import { useFilterContext } from "@calcom/features/insights/context/provider";
+import { useLocale } from "@calcom/lib/hooks/useLocale";
+import { Button, Icon, Tooltip, DataTableFilters } from "@calcom/ui";
+
+import { BookingStatusFilter } from "../BookingStatusFilter";
+import { DateSelect } from "../DateSelect";
+import { RoutingDownload } from "../Download";
+import { FilterType } from "../FilterType";
+import { RoutingFormFieldFilter } from "../RoutingFormFieldFilter";
+import { RoutingFormFilterList } from "../RoutingFormFilterList";
+import { TeamAndSelfList } from "../TeamAndSelfList";
+import { UserListInTeam } from "../UsersListInTeam";
+
+const ClearFilters = () => {
+  const { t } = useLocale();
+  const { filter, clearFilters } = useFilterContext();
+  const { selectedFilter } = filter;
+
+  if (!selectedFilter || selectedFilter?.length < 1) return null;
+
+  return (
+    <Tooltip content={t("clear_filters")}>
+      <Button
+        variant="icon"
+        color="secondary"
+        target="_blank"
+        rel="noreferrer"
+        className="min-w-24 h-[38px] border-0"
+        onClick={() => {
+          clearFilters();
+        }}>
+        <Icon name="x" className="mr-1 h-4 w-4" />
+        {t("clear")}
+      </Button>
+    </Tooltip>
+  );
+};
+
+export const RoutingInsightsFilters = ({ table }: { table: RoutingFormTableType }) => {
+  const { filter } = useFilterContext();
+  const { selectedFilter } = filter;
+
+  // Get all filters that relate to the routing form
+  const routingFormFieldIds = selectedFilter
+    ? selectedFilter.map((filter) => {
+        if (filter.startsWith("rf_")) return filter.substring(3);
+      })
```

Comment:
nit: we could return a filtered array with truthy items only, and don't care at the return statement below